mdm/common/src/main/java/com/gentics/mesh/core/data/dao/ElementResolvingRootDao.java

```diff
@@ -0,0 +1,50 @@
+package com.gentics.mesh.core.data.dao;
+
+import static com.gentics.mesh.core.rest.error.Errors.error;
+import static io.netty.handler.codec.http.HttpResponseStatus.BAD_REQUEST;
+
+import java.util.Stack;
+
+import com.gentics.mesh.core.data.HibBaseElement;
+import com.gentics.mesh.core.data.HibCoreElement;
+import com.gentics.mesh.core.rest.common.RestModel;
+
+/**
+ * An extension to {@link RootDao} with the possibilities to reference persisted leaf entities upon the root entity.
+ * 
+ * @author plyhun
+ *
+ * @param <R>
+ * @param <L>
+ */
+public interface ElementResolvingRootDao<R extends HibCoreElement<? extends RestModel>, L extends HibCoreElement<? extends RestModel>> extends RootDao<R, L> {
```

Comment:
There are implementations (like TagDaos) that implement RootDao but do not implement PersistingRootDao, as the ability to connect leaf entities to the root is optional. Though their leaves still have to be referenced through the root, so technically this element resolving implementation might have been settled in RootDao. But since we do not expose any default implementations in the API, I had to introduce this extra extension in the common code, to share the element resolving implementation between PersistingRootDao implementors and RootDao (not persisting) implementors.